src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -493,6 +493,82 @@ trait GoldTransforms extends SparkSessionWrapper {
 
   }
 
+
+  protected def buildNotebookCommands(
+                                   notebook: PipelineTable,
+                                   clsfIncrementalDF : DataFrame,
+                                 )(auditIncrementalDF: DataFrame): DataFrame = {
+
+
+    val notebookLookupTSDF = notebook.asDF
+      .select("organization_id", "notebook_id", "notebook_path", "notebook_name", "unixTimeMS","date")
+      .withColumnRenamed("notebook_id", "notebookId")
+      .filter('notebookId.isNotNull)
+      .distinct
+      .toTSDF("unixTimeMS", "organization_id","date","notebookId")
+
+    val clsfLookupTSDF = clsfIncrementalDF
+      .select(
+        "organization_id", "state_start_date", "unixTimeMS_state_start", "cluster_id", "cluster_name",
+        "custom_tags", "node_type_id", "current_num_workers", "uptime_in_state_H", "total_cost")
+      .distinct
+      .withColumnRenamed("unixTimeMS_state_start", "unixTimeMS")
+      .withColumnRenamed("cluster_id", "clusterId")
+      .withColumnRenamed("current_num_workers", "node_count")
+      .withColumn("totalCostPMS", col("total_cost") / col("uptime_in_state_H")/lit(3600))
+      .toTSDF("unixTimeMS", "organization_id", "clusterId")
+
+    val colNames: Array[Column] = Array(
+      'organization_id,
+      'clusterId.alias("cluster_id"),
+      'notebookId.alias("notebook_id"),
+      'workspace_name,
+      'date,
+      'timestamp,
+      'notebook_path,
+      'notebook_name,
+      'commandId.alias("command_id"),
+      'commandText.alias("command_text"),
+      'executionTime.alias("execution_time_s"),
+      'sourceIpAddress.alias("source_ip_address"),
+      'userIdentity.alias("user_identity"),
+      'estimated_dbu_cost,
+      'status,
+      'cluster_name,
+      'custom_tags,
+      'node_type_id,
+      'node_count,
+      'response,
+      'userAgent.alias("user_agent"),
+      'unixTimeMS
+    )
+
+    val notebookCodeAndMetaDF = auditIncrementalDF
+      .filter(col("serviceName") === "notebook" && col("actionName") === "runCommand")
+      .selectExpr("*", "requestParams.*").drop("requestParams")
+      .withColumnRenamed("timestamp", "unixTimeMS")
+      .withColumn("timestamp", from_unixtime(col("unixTimeMS") / 1000.0).cast("timestamp"))
+      .withColumn("executionTime",col("executionTime").cast("double"))
+      .filter('notebookId.isNotNull)
+      .toTSDF("unixTimeMS", "organization_id", "date","notebookId")
+      .lookupWhen(
+        notebookLookupTSDF,
+        maxLookAhead = 100
+      )
+      .df
+      .toTSDF("unixTimeMS", "organization_id", "clusterId")
+      .lookupWhen(
+        clsfLookupTSDF,
+        tsPartitionVal = 8
+      )
```

Comment:
this is not the right way to derive this. 

Imagine you have a command that runs for 2 hours and the cluster changes states 4 times during that time...the cost per second will be different at different states throughout the runtime of this cmd execution. You need to complete something like what is below to get the cost per run state. I will DM you a link to a video that will help you understand how to build and work with runstates.

See WorkflowTransforms.scala funcs `jrcpDeriveRunsByClusterState` & `jrcpDeriveCumulativeRuntimeByRunState`.  You need to calculate the costs at the various states of the cluster across the cmdExecution, aggregate the per milli cost across each state to calculate the cost / coreMillis of a cmd execution.

Currently, your logic simply grabs the state at the start of the command but doesn't account for state changes during cmd execution.